src/common/mfu_flist_copy.c

```diff
@@ -993,7 +1024,7 @@ static int mfu_create_directory(
      * creating / striping files in the directory */
 
     /* copy extended attributes on directory */
-    if (copy_opts->preserve) {
+    if (copy_opts->copy_xattrs) {
```

Comment:
It seems like this condition might always be true, since I think ``XATTR_COPY_NONE=1`` in the enum.  I see the function handles NONE correctly, though it might best to avoid calling all together when disabled.  Maybe replace this with ``if (copy_opts->copy_xattrs != XATTR_COPY_NONE)``?

Reply:
Yes, good catch.  I forgot to change it after I switched copy_xattrs from a boolean to an enum, but the mistake was hidden because the function handled NONE correctly.

I changed that condition there and in the other two places mfu_copy_xattrs() is called as you suggested.  I also simplified the logic within mfu_copy_xattrs(), and then tested to confirm it still worked.